src/mscorlib/shared/System/Threading/ExecutionContext.cs

```diff
@@ -21,42 +21,26 @@ namespace System.Threading
 {
     public delegate void ContextCallback(Object state);
 
-    internal struct ExecutionContextSwitcher
-    {
-        internal ExecutionContext m_ec;
-        internal SynchronizationContext m_sc;
-
-        internal void Undo(Thread currentThread)
-        {
-            Debug.Assert(currentThread == Thread.CurrentThread);
-
-            // The common case is that these have not changed, so avoid the cost of a write if not needed.
-            if (currentThread.SynchronizationContext != m_sc)
-            {
-                currentThread.SynchronizationContext = m_sc;
-            }
-
-            if (currentThread.ExecutionContext != m_ec)
-            {
-                ExecutionContext.Restore(currentThread, m_ec);
-            }
-        }
-    }
-
     public sealed class ExecutionContext : IDisposable, ISerializable
     {
-        internal static readonly ExecutionContext Default = new ExecutionContext();
+        internal static readonly ExecutionContext Default = new ExecutionContext(isDefault: true);
 
         private readonly IAsyncLocalValueMap m_localValues;
         private readonly IAsyncLocal[] m_localChangeNotifications;
         private readonly bool m_isFlowSuppressed;
+        private readonly bool m_isDefault;
 
         private ExecutionContext()
         {
             m_localValues = AsyncLocalValueMap.Empty;
             m_localChangeNotifications = Array.Empty<IAsyncLocal>();
         }
 
+        private ExecutionContext(bool isDefault) : base()
```

Comment:
Why is the explicit base ctor call needed?

Reply:
Is private .ctor so can drop the other .ctor w/ init of `m_localValues` and `m_localChangeNotifications` as setting values now deals with them being `null`

Reply:
I don't understand. `private ExecutionContext(bool isDefault) { }` is the same as `private ExecutionContext(bool isDefault) : base() { }`.  You're saying there's a difference?  They should both be making a call to the base ctor().

Reply:
TIL; I think I've been writing unnecessarily verbose code. However don't need to initialize `m_localValues` or `m_localChangeNotifications` so can drop the paramless .ctor